lib/grammar.pegjs

```diff
@@ -145,6 +149,26 @@ param_direction = 'in' __ 'req' { return 'in req'; } / 'in' __ 'opt' { return 'i
     'out' { return 'out'; }
 
 
+// Dataset & Thingpedia Definition
+dataset_name = ident / name:class_name { return '@' + name; }
+
+dataset = _ 'dataset' _ name:dataset_name _ lan:('language' _ literal_string _)? '{' _ examples:(_ example _)* _ '}' {
+    return new Ast.Dataset(name, lan? lan[2] : 'en' , take(examples, 1));
+}
```

Comment:
Should we give structured metadata to datasets? Most other stuff has it, so for consistency I feel like we should.

Also, classes probably should have metadata (which we overlooked in the class merge) to cover User Visible Name and User Visible Description.

Reply:
What kind of metadata do we need? size, description maybe, contributors, or device used?

Reply:
Yeah it could be that too (see also https://github.com/Stanford-Mobisocial-IoT-Lab/almond-cloud/issues/24 ). 
I would keep it open ended though, like we do for functions.

Reply:
so just an object and leave it empty for new dataset?